### PR TITLE
Added new Items model for PC2.0

### PR DESCRIPTION
**Pull request by sandervankasteel**

### CODE REVIEW

File: src/Resources/Items.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+{
+
+    /**
+     * Get all items (previously known as plans) with possible filtering
+     *
+     * @return Collection
+     */
+    public function all(): Collection
+    {
+        $all = Item::all($this->filter);
```
Comment (santal1981) [inline src/Resources/Items.php]:
```suggestion
        $items = Item::all($this->filter);
```

File: src/Resources/Items.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        return collect($all)->map->item();
```
Comment (santal1981) [inline src/Resources/Items.php]:
```suggestion
        return collect($items)->map->item();
```

File: src/Resources/Items.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * Find an item by item_id (key)
+     *
+     * @param string $itemId
+     * @return Item|null
+     */
+    public function findByPlanId(string $itemId): ?Item
+    {
+        return $this->where('item_id', $itemId)->get()->first();
```
Comment (santal1981) [inline src/Resources/Items.php]: get()->first>?

File: src/Resources/Items.php (hunk 1 above)
Comment (santal1981) [reply src/Resources/Items.php]: Align to the content

File: src/Resources/Items.php (hunk 2 above)
Comment (santal1981) [reply src/Resources/Items.php]: Align to the content

**Review by rvanasperen — APPROVED** [review]

**Pull request merged**